spec/ParseWebSocketServer.spec.js

```diff
@@ -75,6 +76,35 @@ describe('ParseWebSocketServer', function () {
     expect(wssError).toBe('Invalid Packet');
   });
 
+  it('closes interrupted connection', function (done) {
+    const onConnectCallback = jasmine.createSpy('onConnectCallback');
+    const http = require('http');
+    const server = http.createServer();
+    const parseWebSocketServer = new ParseWebSocketServer(server, onConnectCallback, {
+      websocketTimeout: 5,
+    }).server;
+    const ws = new EventEmitter();
+    ws.readyState = 0;
+    ws.OPEN = 0;
+    ws.ping = jasmine.createSpy('ping');
+    ws.terminate = jasmine.createSpy('terminate');
+
+    parseWebSocketServer.onConnection(ws);
+
+    // Make sure callback is called
+    expect(onConnectCallback).toHaveBeenCalled();
+    // Make sure we ping to the client
+    setTimeout(function () {
```

Comment:
use `await new Promise(resolve => setTimeout(resolve, 10))`, remove `done` function and callback

Reply:
If I understand correctly you mean is that:
`await new Promise(resolve => {`
`setTimeout(()=>{`
`expect(ws.terminate).toHaveBeenCalled();`
`server.close();`
`resolve();`
`}, 10);`
`});`

Reply:
```
expect(onConnectCallback).toHaveBeenCalled();
await new Promise(resolve => setTimeout(resolve, 10));
expect(ws.ping).toHaveBeenCalled();
await new Promise(resolve => setTimeout(resolve, 10));
expect(ws.terminate).toHaveBeenCalled();
server.close();
```

Reply:
thanks for your help, Done :)